### PR TITLE
feat: sync from v8.26.0 to v8.30.0

### PR DESCRIPTION
**Pull request by kecrily**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What changes did you make? (Give an overview)

sync from v8.26.0 to v8.30.0

### CODE REVIEW

**Review by amareshsm — COMMENTED** [review]
Just a suggestion: Maybe you could have raised it as two separate PRs (removing layout + others, and absolute to relative URL changes). Boz short PRs are simpler to review.

File: src/developer-guide/contributing/new-rules.md [hunk 1]
```diff
@@ -23,7 +22,7 @@ ESLint 是关于规则的。在项目的大部分时间里，我们已经有了
 
 ## 提议一条规则
 
-如果你想提出一条新的规则，请参考如何 [创建拉动请求](/docs/developer-guide/contributing/pull-requests) 或通过填写 [新规则模板](https://github.com/eslint/eslint/issues/new/choose) 提交一个问题。
+如果你想提出一条新的规则，请参考如何 [创建拉动请求](../developer-guide/contributing/pull-requests) 或通过填写 [新规则模板](https://github.com/eslint/eslint/issues/new/choose) 提交一个问题。
```
Comment (mdjermanovic) [inline src/developer-guide/contributing/new-rules.md]:
```suggestion
如果你想提出一条新的规则，请参考如何 [创建拉动请求](pull-requests) 或通过填写 [新规则模板](https://github.com/eslint/eslint/issues/new/choose) 提交一个问题。
```

File: src/developer-guide/contributing/reporting-bugs.md [hunk 2]
```diff
@@ -1,10 +1,9 @@
 ---
 title: Reporting Bugs
-layout: doc
 
 ---
 
-如果你认为你发现了 ESLint 中的漏洞，请在 GitHub 上[创建新议题](https://github.com/eslint/eslint/issues/new/choose)或[拉动请求](/docs/developer-guide/contributing/pull-requests)。
+如果你认为你发现了 ESLint 中的漏洞，请在 GitHub 上[创建新议题](https://github.com/eslint/eslint/issues/new/choose)或[拉动请求](../developer-guide/contributing/pull-requests)。
```
Comment (mdjermanovic) [inline src/developer-guide/contributing/reporting-bugs.md]:
```suggestion
如果你认为你发现了 ESLint 中的漏洞，请在 GitHub 上[创建新议题](https://github.com/eslint/eslint/issues/new/choose)或[拉动请求](pull-requests)。
```

File: src/developer-guide/contributing/working-on-issues.md [hunk 3]
```diff
@@ -1,6 +1,5 @@
 ---
 title: Working on Issues
-layout: doc
```
Comment (mdjermanovic) [inline src/developer-guide/contributing/working-on-issues.md:3]: This file seems to be missing some changes from https://github.com/eslint/eslint/compare/v8.25.0...v8.30.0

**Comment by kecrily** [comment]
> Just a suggestion: Maybe you could have raised it as two separate PRs (removing layout + others, and absolute to relative URL changes). Boz short PRs are simpler to review.

I have made PRs possibly to cover a single piece of content, and then nzakas suggested that I reduce the number of PRs.

Looks like I'll have to reassess the fit, don't worry I'll improve.

File: src/developer-guide/nodejs-api.md [hunk 4]
```diff
@@ -917,14 +916,14 @@ ruleTester.run("my-rule", myRule, {
 
 ---
 
-[configuration object]: ../user-guide/configuring/
-[builtin-formatters]: https://eslint.org/docs/user-guide/formatters/
+[configuration object]: ../user-guide/configuring/index
```
Comment (mdjermanovic) [inline src/developer-guide/nodejs-api.md]: This URL shouldn't have `index`. Also, `[configuration object]` link is unused on the translated page because the text is translated so it appears as `[配置对象]` (check `options.baseConfig` and `options.overrideConfig` list items) so we should either add the URL there or update the text here.

File: src/developer-guide/nodejs-api.md [hunk 5]
```diff
@@ -917,14 +916,14 @@ ruleTester.run("my-rule", myRule, {
...
-[editinfo]: #-editinfo-type
-[loadedformatter]: #-loadedformatter-type
+[eslint-lintfiles]: #◆-eslintlintfilespatterns
+[eslint-linttext]: #◆-eslintlinttextcode-options
+[eslint-loadformatter]: #◆-eslintloadformatternameorpath
+[lintresult]: #◆-lintresult-类型
+[lintmessage]: #◆-lintmessage-类型
+[suppressedlintmessage]: #◆-suppressedlintmessage-类型
+[editinfo]: #◆-editinfo-类型
+[loadedformatter]: #◆-loadedformatter-类型
```
Comment (mdjermanovic) [inline src/developer-guide/nodejs-api.md]: Why this change? It doesn't seem to work with `◆`.

File: src/developer-guide/selectors.md [hunk 6]
```diff
@@ -138,4 +137,16 @@ module.exports = {
 
 ### 已知问题
 
-由于 [esquery](https://github.com/estools/esquery) 中的一个 [bug](https://github.com/estools/esquery/issues/68)，包含正斜线字符 `/` 的正则表达式不能被正确解析，所以 `[value=/some/path/]` 会出现语法错误。你可以将 `/` 字符替换成其对应的 unicode 字符，像这样 `[value=/some\u002Fpath/]`，来[解决这个问题](https://github.com/estools/esquery/issues/68)。
+由于 [esquery](https://github.com/estools/esquery) 中的一个 [bug](https://github.com/estools/esquery/issues/68)，无法正确解析包含正斜杠字符 `/` 的正则表达式，所以 `[value=/some/path/]` 会出现语法错误。你可以将 `/` 字符替换成其对应的 unicode 字符，像这样 `[value=/some\u002Fpath/]`，来[解决这个问题](https://github.com/estools/esquery/issues/68)。
```
Comment (mdjermanovic) [inline src/developer-guide/selectors.md]:
```suggestion
由于 [esquery](https://github.com/estools/esquery) 中的一个 [bug](https://github.com/estools/esquery/issues/68)，无法正确解析包含正斜杠字符 `/` 的正则表达式，所以 `[value=/some\/path/]` 会出现语法错误。你可以将 `/` 字符替换成其对应的 unicode 字符，像这样 `[value=/some\u002Fpath/]`，来[解决这个问题](https://github.com/estools/esquery/issues/68)。
```

File: src/rules/no-unused-vars.md [hunk 7]
```diff
@@ -125,7 +124,7 @@ var global_var = 42;
 ```json
 {
     "rules": {
-        "no-unused-vars": ["error", { "vars": "all", "args": "after-used", "ignoreRestSiblings": false }]
+        "no-unused-vars": ["error", { "vars": "all", "args": "after-used", "": false }]
```
Comment (mdjermanovic) [inline src/rules/no-unused-vars.md]: This was probably accidentally removed.

File: src/rules/prefer-const.md [hunk 8]
```diff
@@ -94,7 +93,7 @@ console.log(a);
 
 // `i` gets a new binding each iteration
 for (const i in [1, 2, 3]) {
-  console.log(i);
+  console.log(a);
```
Comment (mdjermanovic) [inline src/rules/prefer-const.md]: Weird, in the original document it used to be `a` and now it's changed to `i`, but this change seems to be doing the opposite.

File: src/user-guide/configuring/language-options.md [hunk 9]
```diff
@@ -187,12 +188,11 @@ And in YAML:
...
 
 请注意，支持 JSX 语法并不等同于支持 React。React 对 JSX 语法应用了特定的语义，而 ESLint 并不能识别。如果你使用 React 并且使用 React 语法，我们建议使用 [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react)。
-同样地，支持 ES6 语法不等于支持新的 ES6 全局变量（例如，新的类型，如 `Set`）。
-使用 `{ "parserOptions": { "ecmaVersion": 6 } }`启用 ES6 语法；使用 `{ "env": { "es6": true } }`. `{ "env": { "es6": true }}` 启用新的 ES6 全局变量；`{ "parserOptions": { "ecmaVersion": 6 } }` 会自动启用 ES6 语法，但不会自动启用 ES6 全局变量。
+同样地，支持 ES6 语法不等于支持新的 ES6 全局变量（例如，新的类型，如 `Set`）。使用 `{ "parserOptions": { "ecmaVersion": 6 } }` 启用 ES6 语法；使用 `{ "env": { "es6": true } }` 启用新的 ES6 全局变量，设置 `{ "env": { "es6": true }}` 会自动启用 ES6 语法，但 `{ "parserOptions": { "ecmaVersion": 6 } }` 则不会自动启用 ES6 全局变量。
 
 可以在 `.eslintrc.*` 文件中通过使用 `parserOptions` 属性设置解析器选项。可用选项有：
 
-*  `ecmaVersion`  - 设置为 3、5（默认）、6、7、8、9、10、11、12 或 13，以指定你要使用的 ECMAScript 语法的版本。你也可以设置为 2015（6）、2016（7）、2017（8）、2018（9）、2019（10）、2020（11）、2021（12）或 2022（13）来使用基于年份的命名。你也可以设置 `lastest` 来使用受支持的最新版本。
+* `ecmaVersion`  - 设置为 3、5（默认）、6、7、8、9、10、11、12 或 13，以指定你要使用的 ECMAScript 语法的版本。你也可以设置为 2015（6）、2016（7）、2017（8）、2018（9）、2019（10）、2020（11）、2021（12）或 2022（13）来使用基于年份的命名。你也可以设置 `"lastest"` 来使用受支持的最新版本。
```
Comment (mdjermanovic) [inline src/user-guide/configuring/language-options.md]:
```suggestion
* `ecmaVersion`  - 设置为 3、5（默认）、6、7、8、9、10、11、12 或 13，以指定你要使用的 ECMAScript 语法的版本。你也可以设置为 2015（6）、2016（7）、2017（8）、2018（9）、2019（10）、2020（11）、2021（12）或 2022（13）来使用基于年份的命名。你也可以设置 `"latest"` 来使用受支持的最新版本。
```

File: src/user-guide/core-concepts.md [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+
+更多信息请参见[使用可共享配置包](./configuring/configuration-files#使用可共享配置包)。
+
+## 插件
+
+ESLint 插件是一个包含 ESLint 规则、配置、解析器和环境变量的集合的 npm 模块。通过插件包括自定义规则。插件可以强制使用某个风格指南并支持 JavaScript 扩展（比如 TypeScript）、库（比如 React）和框架（比如 Angular）。
+
+插件的流行用例就是强制执行框架规定的最佳实践。比如 [@angular-eslint/eslint-plugin](https://www.npmjs.com/package/@angular-eslint/eslint-plugin) 包括了使用 Angular 框架的最佳实践。
+
+更多信息请参见[配置插件](./configuring/plugins.md)。
```
Comment (mdjermanovic) [inline src/user-guide/core-concepts.md]:
```suggestion
更多信息请参见[配置插件](./configuring/plugins)。
```

File: src/user-guide/core-concepts.md [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+
+ESLint 处理器可以从其他类型的文件中提取 JavaScript 代码，然后让 ESLint 对 JavaScript 代码进行检查。另外，你也可以在用 ESLint 解析 JavaScript 代码之前使用一个处理器对其进行处理。
+
+例如 [eslint-plugin-markdown](https://github.com/eslint/eslint-plugin-markdown) 就包括一个自定义处理器，让你可以对 Markdown 代码块内的 JavaScript 代码进行检查。
+
+## 格式化工具
+
+ESLint 格式化工具决定了命令行检查结果的样子。
+
+更多信息请参见[格式化工具](./formatters)。
```
Comment (mdjermanovic) [inline src/user-guide/core-concepts.md]:
```suggestion
更多信息请参见[格式化工具](./formatters/)。
```

File: src/developer-guide/package-json-conventions.md [hunk 12]
```diff
@@ -36,6 +35,10 @@ Scripts that generate a set of files from source code and / or data MUST have na
 
 If a package contains any `build:*` scripts, there MAY be a script named `build`. If so, SHOULD produce the same output as running each of the `build` scripts individually. It MUST produce a subset of the output from running those scripts.
 
+### Release
+
+Scripts that have public side effects (publishing the web site, committing to Git, etc.) MUST begin with `release`.
```
Comment (amareshsm) [inline src/developer-guide/package-json-conventions.md]:
```suggestion
Scripts that have public side effects (publishing the website, committing to Git, etc.) It MUST begin with `release`.
```

File: src/rules/no-unused-vars.md [hunk 13]
```diff
@@ -394,6 +334,66 @@ try {
...
+_o = 1;
+[_o, p] = foo;
+p;
+```
+
+:::
+
+### ignoreRestSiblings
+
+`ignoreRestSiblings`选项是一个布尔值（默认：`false`）。使用 [Rest Property](https://github.com/tc39/proposal-object-rest-spread) 可以从一个对象的 `omit` 属性，但默认情况下，兄弟姐妹的属性被标记为 `unused`。启用这个选项后，其余属性的兄弟姐妹将被忽略。
```
Comment (amareshsm) [inline src/rules/no-unused-vars.md]:
```suggestion
`ignoreRestSiblings` 选项是一个布尔值（默认：`false`）。使用 [Rest Property](https://github.com/tc39/proposal-object-rest-spread) 可以从一个对象的 `omit` 属性，但默认情况下，兄弟姐妹的属性被标记为 `unused`。启用这个选项后，其余属性的兄弟姐妹将被忽略。
```

File: src/user-guide/command-line-interface.md [hunk 14]
```diff
@@ -150,8 +165,15 @@ npx eslint --env browser --env node file.js
...
 
-此项用于指定 ESLint 在目录中匹配目标文件所用扩展。
-默认情况下，ESLint 会搜索 `*.js` 文件及与配置中 `overrides` 条目所匹配的文件。
+此选项用于指定 ESLint 在目录中匹配目标文件所用扩展。
+
+* **参数类型**：字符串。文件扩展名。
+* **多个参数**：支持
+* **默认值**：`.js` 和与配置中 `overrides` 条目相匹配的文件。
+
+只有在检查模式为目录时才会使用 `--ext`。如果你使用 glob 模式或文件名，那么会忽略 `--ext`。例如，`npx eslint "lib/*" --ext .js` 会检查`lib/` 目录下的所有文件，不管扩展名是什么。
```
Comment (amareshsm) [inline src/user-guide/command-line-interface.md]:
```suggestion
只有在检查模式为目录时才会使用 `--ext`。如果你使用 glob 模式或文件名，那么会忽略 `--ext`。例如，`npx eslint "lib/*" --ext .js` 会检查`lib/` 目录下的所有文件，不管扩展名是什么。
```
```suggestion
只有在检查模式为目录时才会使用 `--ext`。如果你使用 glob 模式或文件名，那么会忽略 `--ext`。例如，`npx eslint "lib/*" --ext .js` 会检查 `lib/` 目录下的所有文件，不管扩展名是什么。
```

File: src/user-guide/getting-started.md [hunk 15]
```diff
@@ -54,7 +73,7 @@ yarn run eslint yourfile.js
 }
 ```
 
-`"semi"` 和 `"quotes"` 是 ESLint [规则](/docs/rules)的名称。第一个值代表规则的错误级别，有以下几种可供选择：
+`"semi"` 和 `"quotes"` 是 ESLint [规则](../rules)的名称。第一个值代表规则的错误级别，有以下几种可供选择：
```
Comment (amareshsm) [inline src/user-guide/getting-started.md]:
```suggestion
`"semi"` 和 `"quotes"` 是 ESLint [规则](../rules/)的名称。第一个值代表规则的错误级别，有以下几种可供选择：
```

File: src/user-guide/getting-started.md [hunk 16]
```diff
@@ -70,7 +89,7 @@ yarn run eslint yourfile.js
 }
 ```
 
-这一行将启用[所有标记为“推荐”的规则](/docs/rules)。另外，你也可以通过在 [npmjs.com](https://www.npmjs.com/search?q=eslint-config) 上搜索“eslint-config”并使用别人创建的配置。在没有扩展别人的可共享配置或在配置中明确启用规则时，ESLint 不会限制你的代码。
+这一行将启用[所有标记为“推荐”的规则](../rules)。另外，你也可以通过在 [npmjs.com](https://www.npmjs.com/search?q=eslint-config) 上搜索“eslint-config”并使用别人创建的配置。在没有扩展别人的可共享配置或在配置中明确启用规则时，ESLint 不会限制你的代码。
```
Comment (amareshsm) [inline src/user-guide/getting-started.md]:
```suggestion
这一行将启用[所有标记为“推荐”的规则](../rules/)。另外，你也可以通过在 [npmjs.com](https://www.npmjs.com/search?q=eslint-config) 上搜索“eslint-config”并使用别人创建的配置。在没有扩展别人的可共享配置或在配置中明确启用规则时，ESLint 不会限制你的代码。
```

File: src/developer-guide/nodejs-api.md (hunk 5 above)
Comment (kecrily) [reply src/developer-guide/nodejs-api.md]: VS Code's Markdown Fragment Links check reminds me to do this, but it seems to be an error, and I'll revert it back.

Comment (kecrily) [reply src/developer-guide/nodejs-api.md]: I just found out that the anchor generated by eleventy does not support Chinese, which means that all the Chinese anchors used before are invalid. I may need to figure out how to fix it, but it won't be in this PR

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks! Leaving open for @amareshsm to verify.

**Review by amareshsm — APPROVED** [review]
LGTM
